doc/mtransfer.rst

```diff
@@ -0,0 +1,123 @@
+.. _mtransfer:
+
+=========
+mtransfer
+=========
+
+``mtransfer`` allows WiredTiger databases to be exported from one MongoDB
+instance and imported into another.
+
+
+Caveats
+~~~~~~~
+
+The ``mtransfer`` script is EXPERIMENTAL and has a number of important usage caveats:
+
+- MongoDB must be started with the ``--directoryperdb`` flag.
+- ``mtransfer`` does not work with sharding, the encrypted storage engine, or
+  MMAPv1 data files.
+- To export or import a database, MongoDB must not be running using either the
+  source or destination database paths.
+- A database must be imported to all nodes in a replica set or query results
+  will be inconsistent.
+- A database cannot be imported to any node in the replica set it was exported
+  from. Collections have unique identifiers, and this would violate that uniqueness.
+- The `wiredtiger` Python library must include support for any compression formats
+  uses by your MongoDB deployment (i.e. ``snappy``. ``zlib``, ``zstd``). If you are
+  encountering errors using the ``wiredtiger`` module as installed via `pip`,
+  you may need to `Build and install WiredTiger from source
+  <http://source.wiredtiger.com/develop/build-posix.html>`__.
+
+While there are some sanity checks built into the script, manipulating MongoDB
+files directly is inherently dangerous. Take care to test and backup your data.
+
+Usage
+~~~~~
+
+.. code-block:: bash
+
+  mtransfer [-h] [--version] [--verbose] [--dbpath DBPATH]
+                    {export,import} database
+
+General Parameters
+~~~~~~~~~~~~~~~~~~
+
+Help
+----
+``-h, --help``
+   shows the help text and exits.
+
+Version
+-------
+``--version``
+   shows the version number and exits.
+
+Verbosity
+---------
+``--verbose``
+   shows extra information
+
+Commands
+~~~~~~~~
+
+``mtransfer`` reads or writes the data files in a MongoDB instance
+using the WiredTiger storage engine.
+
+Database Path
+-------------
+``--dbpath path`` the path to the MongoDB database files (defaults to
+the current working directory where the script is run).
+
+Command
+-------
+``export``
+creates a file ``mtransfer.bson`` in the specified database directory.
+This must be copied along with the other files in the directory to the
+destination server(s).
+
+``import`` reads the ``mtransfer.bson`` file from the specified database
+directory and updates MongoDB's metadata to include the database.
+
+Database
+---------
+The name of the database to export / import.  The MongoDB database name
+must match the directory name on disk for the export, and the MongoDB
+database name will be set to the directory name for the import.
+
+Example
+~~~~~~~
+
+On the origin
+-------------
+
+Before starting, ensure MongoDB is not running.
+
+.. code-block:: bash
```

Comment:
I'd be tempted to turn this code block into a numbered procedure.

1. Change into the origin directory and create 'olddb/mtransfer.bson' with exported metadata:
   ```
   cd /from/dbpath
   mtransfer export olddb
   ```
2.  ...

Reply:
@kallimachos Good suggestion; pushed a commit which converts the overall example into a single procedure with numbered steps.